core/src/commonMain/kotlin/com/powersync/PowerSyncDatabase.kt

```diff
@@ -134,6 +134,25 @@ public interface PowerSyncDatabase : Queries {
     @Throws(PowerSyncException::class, CancellationException::class)
     public suspend fun getNextCrudTransaction(): CrudTransaction?
 
+    /**
+     * Get a batch of crud data from multiple transactions to upload.
+     *
+     * Returns null if there is no data to upload.
+     *
+     * Use this from the [PowerSyncBackendConnector.uploadData]` callback.
+     *
+     * Once the data have been successfully uploaded, call [CrudBatch.complete] before
+     * requesting the next batch.
+     *
+     * Unlike [getCrudBatch], this groups data by transaction, allowing developers to
+     * upload multiple complete transactions in a single batch operation.
+     *
+     * @param transactionLimit The maximum number of transactions to include in the batch.
+     * Default is 5.
+     */
+    @Throws(PowerSyncException::class, CancellationException::class)
+    public suspend fun getNextCrudTransactionBatch(transactionLimit: Int = 10): CrudBatch?
```

Comment:
Another thing we would likely want here is to have an optional limit on the amount of crud items - here it can essentially be unbounded if you have multiple huge transactions.

I'm not sure yet if we'll need both limits or if a limit on items is enough. Basically the behavior would be to return the maximum amount of transactions so that the total items in those transactions doesn't exceed the limit, but also return at least one transaction.